File: bonassa-webapp/src/main/webapp/app/scripts/services/parameterService.js
'use strict';

/**
 * @ngdoc service
 * @name webappApp.parameterService
 * @description # parameterService Service in the webappApp.
 */
angular.module('webappApp').service('parameterService',
		[ '$http', function($http) {
			var doRequest = function(url, params, method) {
				if (params != null) {
					if (method == 'GET') {
						return  $http({
							method : method,
							url : url,
							params : params
						});
					} else {
						return $http({
							method : method,
							url : url,
							data : params
						});
					}
				} else {
					return $http({
						method : method,
						url : url
					});
				}
			}
			return {
				getParams : function() {
					return doRequest('../rest/parameter/list', null, 'GET');
				},
				insertValue : function(param, value) {
					var params = {};
					params.paramId = param.id;
					if (value.parent != null)
						params.parentId = value.parent.id;
					params.value = value.value;
					console.log(params);
					
					return doRequest('../rest/parameter/insertValue', params, 'GET');
				},
				updateValue : function(param, value) {
					var params = {};
					params.paramId = value.id;
					if (value.parent != null)
						params.parentId = value.parent.id;
					params.value = value.value;
					console.log(params);
					return doRequest('../rest/parameter/updateValue', params, 'GET');
				}
			}
		} ]);

File: bonassa-core/src/main/java/com/esense/bonassa/core/service/ParameterService.java
package com.esense.bonassa.core.service;

import java.util.List;

import com.esense.bonassa.core.entity.Parameter;

public interface ParameterService {

	public List<Parameter> list();

	public void insert(Parameter param);

	public void insertValue(Integer paramId, Integer parentId, String value);

	public void updateValue(Integer paramId, Integer parentId, String value);

	public Parameter selectByName(String paramName);
	
	public Parameter selectById(Integer id);

//	public Parameter selectByParentValue(Integer parentId);

}
